libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -2348,19 +2355,27 @@ void WaveTrack::Join(
    double t0, double t1, const ProgressReporter& reportProgress)
 {
    assert(IsLeader());
-   const auto channels = TrackList::Channels(this);
-   // Merge all WaveClips overlapping selection into one
+   const auto intervals = Intervals();
+   std::vector<std::shared_ptr<Interval>> intervalsToJoin;
+   for (auto interval : intervals)
+      if (interval->IntersectsPlayRegion(t0, t1))
+         // TODO after this loop's iteration executes, the object referred to by
+         // `interval` would get destroyed if `intervalsToJoin` were a vector of
+         // raw pointers - why is that ?
+         intervalsToJoin.push_back(interval);
    BasicUI::SplitProgress(
-      channels.begin(), channels.end(),
-      [&](WaveTrack* pChannel, ProgressReporter child) {
-         pChannel->JoinOne(*pChannel, t0, t1, child);
+      intervalsToJoin.begin(), intervalsToJoin.end(),
+      [&](const std::shared_ptr<Interval>& interval, ProgressReporter child) {
+         interval->ApplyStretchRatio(child);
       },
       reportProgress);
+
+   for (const auto pChannel : TrackList::Channels(this))
+      JoinOne(*pChannel, t0, t1);
```

Comment:
I see this commit that changes join behavior, and I ask whether it might make unnecessary changes in case clips already have equal stretches.

Reply:
Ah, crap, I forgot that check ... Thank you !